sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ParameterValidationInMethodShouldBeWrapped.cs

```diff
@@ -74,5 +74,18 @@ protected override sealed void Initialize(SonarAnalysisContext context)
                 },
                 RegisterSyntaxKinds);
         }
+
+        private static MethodDeclarationSyntax GetFirstEnclosingMethod(TRegisterExpressionSyntax registerExpression)
```

Comment:
From the name, I wouldn't expect this function to return null for a local function.
Not sure what a good name would be though - _GetMethodIfNotLocalFunction_ is clumsy, but more accurate.